src/d3a/models/state.py

```diff
@@ -46,13 +47,25 @@ def __init__(self):
         self.available_energy_kWh = \
             {slot: 0. for slot in generate_market_slot_list()}
 
+    def get_state(self):
+        return {"available_energy_kWh": convert_pendulum_to_str_in_dict(self.available_energy_kWh)}
+
+    def load_state(self, state_dict):
+        self.available_energy_kWh = state_dict["available_energy_kWh"]
+
```

Comment:
One potential optimisation that I see in terms of memory is rather than saving and retrieving the whole profile is to rather purge the value of market that has already been surpassed as they are no more needed and simply been there now there as a slug and increasing memory requirement.

Reply:
We do it already in the strategies IIRC:
https://github.com/gridsingularity/d3a/blob/119abaaa8776afb9d68568573991a6d807b0c084/src/d3a/models/strategy/load_hours.py#L156

Reply:
Ah yes, ignore my comment.